MovingBlocks/Terasology
# Fix crash when system locale is complex

### Contains

Fixes #5157. Detects if a locale looks like e.g. `en_UA` and converts it to e.g. `en`, preventing a game crash.

Also re-enables the *ancient* part-pirate localization by adding it to the "whitelist" of allowed locales.

### How to test

Build from source with a "weird locale" (I did this on a non-M1/M2 Mac) and verify the game no longer crashes :)
